docs/_drafts/scv2-hodometer.md

```diff
@@ -0,0 +1,313 @@
+The following is what the C4 model calls a **component** diagram, which is about how the modules or services _within_ a container --- an application --- interact with one another and with outside components.
+In the context of Golang, which both metrics systems are written in, I'm choosing to interpret interfaces and important structs --- those with methods defining business logic --- as components.
+
+### Spartakus component view
+
+![Spartakus component architecture](./scv2-hodometer-spartakus-component.png)
+
+Here we can see that the volunteer application, the metrics publisher, has five main components.
+
+The `volunteer` struct encapsulates the overall program logic of scheduling runs and orchestrating the other components that capture and collate the metrics of interest.
+
+The `serverVersioner` and `nodeLister` are interfaces the `volunteer` depends upon.
+As the names suggest, they're there to retrieve information about the k8s server version and details on each node in the cluster respectively.
+As it happens, they are both implemented by the `kubeClientWrapper` struct in [kubernetes.go](https://github.com/kubernetes-retired/spartakus/blob/a593c5e0f13efa415752bf4ebaccdacddf7e979a/pkg/volunteer/kubernetes.go#L138), but this is an implementation detail.
+
+The `extensionsLister` is another interface, which is responsible for returning the key-value pairs of user-defined static data Spartakus calls "extensions".
+While before we had two interfaces and one implementation, this time there's one interface but two implementations!
+The first of these is responsible for transforming a list of bytes (a byte slice in Go nomenclature) into a list of "extensions".
+The other takes care of searching a specified filesystem path for relevant-looking files and passing the contents of these to the aforementioned byte-handling implementation.
+
+Finally, there's a `Database` interface, which provides a unified entrypoint to the various different storage backends Spartakus supports: BigQuery, HTTP endpoints, and `STDOUT`.
+These different implementations all support a simple `Store()` method defined by the `Database` interface.
+
+### Hodometer component view
+
+In contrast, Hodometer has far fewer components and implementations to be aware of.
+It has three main business logic structs: a punctuator, a collector, and a publisher.
+
+![Hodometer component architecture](./scv2-hodometer-hodometer-component.png)
+
+The `Punctuator` struct is responsible for running code on a periodic basis, much like the idea of a punctuator in Kafka Streams, as it happens.
+Unlike the `volunteer` in Spartakus, the `Punctuator` is _not_ responsible for defining what to run.
+Instead, in Hodometer this is provided by the wiring logic in the `main` function, which can be found [here](https://github.com/SeldonIO/seldon-core/blob/d3502062bbbb18a08032201917ceea07e124be41/hodometer/cmd/hodometer/main.go#L70):
+
+```go
+	punctuator.Run(
```

Comment:
:wrench: Although correctly indented for where it was copied from the code, the indentation in these code blocks isn't important for readers of the article.  In fact, given the rendering of tabs can be significant (8 spaces in GitHub), it probably hurts legibility to leave the tabs in place.

As such, the tabs in code blocks should be removed.

Reply:
Fixed in https://github.com/agrski/agrski.github.io/pull/40/commits/748e359c0099285a39080342694949c6478a6da0